### PR TITLE
*: Rename "default" cluster to "quickstart"

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR renames the "default" cluster to "quickstart" for new environments, existing environments will not have their cluster renamed. To keep the default value for the "cluster" variable to "default" we add a Catalog migration that runs for existing environments.

### Motivation

Fixes https://github.com/MaterializeInc/materialize/issues/22351

### Tips for reviewer

There are a large number of changes here, but the majority of them are test-only changes. The PR is broken up into commits that can ideally be review independently:

1. The changes to actually rename the "default" cluster to "quickstart"
2. Updates to sqllogictests
3. Updates to testdrive tests
4. Updates to our python tests
5. Updates to our dbt tests
6. Updates to our Rust tests
7. Updates to the legacy-upgrade tests

The accompanying changes to our user documentation is here: https://github.com/MaterializeInc/materialize/pull/24035

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Renames the cluster created for every environment from "default" to "quickstart"

### CODE REVIEW

**Review by def- — COMMENTED** [review]
Glad to see you triggered nightly already, I'm expected lots of related failures. If that's green, ok

**Review by def- — CHANGES_REQUESTED** [review]
The "[Detect references to already closed issues](https://buildkite.com/materialize/nightlies/builds/5694#_)" can be ignored, but platform-checks looks like a new issue: 
```
Running manipulate() from <materialize.checks.all_checks.identifiers.Identifiers object at 0x7ff8495ce250>
--- docker compose exec -T testdrive testdrive --kafka-addr=kafka:9092 --schema-registry-url=http://schema-registry:8081 --materialize-url=postgres://materialize@materialized:6875 --materialize-internal-url=postgres://materialize@materialized:6877 --aws-endpoint=http://localstack:4566 --no-reset --default-timeout=300s --seed=1 --persist-blob-url=file:///mzdata/persist/blob --persist-consensus-url=postgres://root@materialized:26257?options=--search_path=consensus --var=replicas=1 --var=default-replica-size=4-4 --var=default-storage-size=4-1 --source=/var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-092248735da8c3952-1/materialize/nightlies/misc/python/materialize/checks/all_checks/identifiers.py:150
> SET CLUSTER=identifiers;
rows match; continuing at ts 1703179510.3750253
> SET DATABASE="	 Â…Â áš€â€‚â€ƒâ€‚â€ƒâ€„â€…â€†â€‡â€ˆâ€‰â€Šâ€‹â€¨â€©â€¯âŸã€€";
rows match; continuing at ts 1703179510.3760698
> CREATE MATERIALIZED VIEW "../../../../../../../../../../../etc/hosts"." " IN CLUSTER quickstart AS
SELECT "<foo val=â€œbarâ€ />", c2 as "<foo val=â€œbarâ€ />" FROM "../../../../../../../../../../../etc/hosts"."<a href=""javascript\x0D:javascript:alert(1)"" id=""fuzzelement1"">test</a>";

^^^ +++
4:1: error: executing query failed: db error: ERROR: column "<foo val=â€œbarâ€ />" specified more than once: ERROR: column "<foo val=â€œbarâ€ />" specified more than once
```
I don't see that column name being specified twice!

**Comment by ParkMyCar** [comment]
> Glad to see you triggered nightly already, I'm expected lots of related failures. If that's green, ok

There were a lot of test failures, but I've been iterating :)

> ```
> Running manipulate() from <materialize.checks.all_checks.identifiers.Identifiers object at 0x7ff8495ce250>
> --- docker compose exec -T testdrive testdrive --kafka-addr=kafka:9092 --schema-registry-url=http://schema-registry:8081 --materialize-url=postgres://materialize@materialized:6875 --materialize-internal-url=postgres://materialize@materialized:6877 --aws-endpoint=http://localstack:4566 --no-reset --default-timeout=300s --seed=1 --persist-blob-url=file:///mzdata/persist/blob --persist-consensus-url=postgres://root@materialized:26257?options=--search_path=consensus --var=replicas=1 --var=default-replica-size=4-4 --var=default-storage-size=4-1 --source=/var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-092248735da8c3952-1/materialize/nightlies/misc/python/materialize/checks/all_checks/identifiers.py:150
> > SET CLUSTER=identifiers;
> rows match; continuing at ts 1703179510.3750253
> > SET DATABASE="	� Â…Â áš€â€‚â€ƒâ€‚â€ƒâ€„â€…â€†â€‡â€ˆâ€‰â€Šâ€‹â€¨â€©â€¯â�Ÿã€€";
> rows match; continuing at ts 1703179510.3760698
> > CREATE MATERIALIZED VIEW "../../../../../../../../../../../etc/hosts"." " IN CLUSTER quickstart AS
> SELECT "<foo val=â€œbarâ€� />", c2 as "<foo val=â€œbarâ€� />" FROM "../../../../../../../../../../../etc/hosts"."<a href=""javascript\x0D:javascript:alert(1)"" id=""fuzzelement1"">test</a>";
> 
> ^^^ +++
> 4:1: error: executing query failed: db error: ERROR: column "<foo val=â€œbarâ€� />" specified more than once: ERROR: column "<foo val=â€œbarâ€� />" specified more than once
> ```
> 
> I'm wondering why that is failing now

I'm not sure why this is failing either, I changed some strings to f-strings but not this one. I was going to dig into it

**Comment by ParkMyCar** [comment]
@def- it seems like [these two lines](https://github.com/ParkMyCar/materialize/blob/main/misc/python/materialize/blns.json#L122-L123) have the same exact content, was that intentional?. I copied them into Rust Playground and inspected their byte representation. So it seems _possible_ that this conflict could happen, because `column` and `alias` could map to the same ident. Maybe we're just getting unlucky now?

**Comment by def-** [comment]
Oooh, that is indeed not supposed to happen. Let me scan that file for other duplicates, thanks!

File: misc/python/materialize/checks/checks.py [hunk 1]
```diff
@@ -44,6 +44,15 @@ def _unsafe_schema(self) -> str:
...
             return "mz_internal"
 
+    def _default_cluster(self) -> str:
+        """
+        :return: name of the cluster created in all environments.
+        """
+        if self.base_version >= MzVersion.parse_mz("v0.81.0-dev"):
+            return "quickstart"
+        else:
+            return "default"
```
Comment (jkosh44) [inline misc/python/materialize/checks/checks.py:54]: Nice

File: src/adapter/src/catalog/migrate.rs [hunk 2]
```diff
@@ -324,6 +330,15 @@ fn ast_rewrite_rewrite_type_schemas_0_81_0(stmt: &mut Statement<Raw>) {
     Rewriter.visit_statement_mut(stmt);
 }
 
+fn persist_default_cluster_0_81_0(
+    txn: &mut Transaction<'_>,
+    _conn_catalog: &ConnCatalog<'_>,
+    _connection_context: &ConnectionContext,
+) -> Result<(), anyhow::Error> {
+    txn.upsert_system_config("cluster", "default".to_string())?;
```
Comment (jkosh44) [inline src/adapter/src/catalog/migrate.rs:338]: Do we need to check if they've already overwritten their `cluster` var and not `upsert` in that case?

File: src/adapter/src/catalog/state.rs [hunk 3]
```diff
@@ -220,7 +220,7 @@ impl CatalogState {
             state: Cow::Borrowed(self),
             unresolvable_ids: BTreeSet::new(),
             conn_id: SYSTEM_CONN_ID.clone(),
-            cluster: "default".into(),
+            cluster: "quickstart".into(),
```
Comment (jkosh44) [inline src/adapter/src/catalog/state.rs:223]: Can we use `mz_catalog::durable::initialize::DEFAULT_USER_CLUSTER_NAME` here?

File: src/catalog/tests/snapshots/debug__stash_opened_trace.snap [hunk 4]
```diff
@@ -151,20 +151,22 @@ Trace {
...
-                                        IdNameV1(
-                                            IdNameV1 {
-                                                id: "u1",
-                                                name: "default",
+                                        UpdatePrivilegeV1(
+                                            UpdatePrivilegeV1 {
+                                                object_id: "Du1",
+                                                grantee_id: "p",
+                                                grantor_id: "s1",
+                                                privileges: "U",
```
Comment (jkosh44) [inline src/catalog/tests/snapshots/debug__stash_opened_trace.snap:169]: What happened here? There's a lot more changes than I was expecting.

File: test/cluster/query-without-default-cluster/query-without-default-cluster.td [hunk 5]
```diff
@@ -23,11 +23,11 @@ DROP CLUSTER default CASCADE
...
 ! SELECT * FROM (VALUES (1), (2)) where column1 > 1
-contains:unknown cluster 'default'
+contains:unknown cluster 'quickstart'
 
 ! SELECT generate_series(0, 10)
-contains:unknown cluster 'default'
+contains:unknown cluster 'quickstart'
 
-# to make it easier to run the test again, let's recreate the default cluster
+# to make it easier to run the test again, let's recreate the defaquickstartult cluster
```
Comment (jkosh44) [inline test/cluster/query-without-default-cluster/query-without-default-cluster.td:31]:
```suggestion
# to make it easier to run the test again, let's recreate the quickstart cluster
```

**Review by morsapaes — APPROVED** [review]

File: src/adapter/src/catalog/migrate.rs (hunk 2 above)
Comment (ParkMyCar) [reply src/adapter/src/catalog/migrate.rs:338]: https://github.com/MaterializeInc/materialize/pull/23903 has yet to be included in a release, so it wouldn't have been previously possible for users to overwrite their `cluster` var in this case

File: src/adapter/src/catalog/state.rs (hunk 3 above)
Comment (ParkMyCar) [reply src/adapter/src/catalog/state.rs]: This made me realize we shouldn't use a static value for a session-less user, and instead we should use whatever value is in the system vars, updated!

File: src/catalog/tests/snapshots/debug__stash_opened_trace.snap (hunk 4 above)
Comment (ParkMyCar) [reply src/catalog/tests/snapshots/debug__stash_opened_trace.snap:169]: Turns out the snapshot isn't sorted, so the large amount of changes comes from audit log events being ordered differently in the formatting of the snapshot. If you'd like I can fix the ordering in this PR, or in a follow up!

File: test/cluster/query-without-default-cluster/query-without-default-cluster.td (hunk 5 above)
Comment (ParkMyCar) [reply test/cluster/query-without-default-cluster/query-without-default-cluster.td]: Whoops, updated!

File: src/catalog/tests/snapshots/debug__stash_opened_trace.snap (hunk 4 above)
Comment (jkosh44) [reply src/catalog/tests/snapshots/debug__stash_opened_trace.snap:169]: No that's ok. As long as it's just ordering it's fine.

**Review by jkosh44 — APPROVED** [review]
LGTM

**Review by def- — APPROVED** [review]

**Pull request merged**